File: stack.py
class Stack:
    def __init__(self, size):
        self.array = [None] * size
        self.length = 0
    
    def push(self, data = None):
        if data == None:
            print("Argument shouldn't be empty")
            return
        if self.length == len(self.array):
            print ("Stack Overflow")
            return
        self.array[self.length] = data
        print(self.array)
        self.length += 1

    def pop(self):
        if self.length == 0:
            print("Cannot pop from empty stack")
            return
        self.length -= 1
        x = self.array[self.length]
        self.array[self.length] = None
        print(x)
        print(self.array)
        return self.array[self.length]

    def print_stack(self):
        print (self.array)

    def peek(self):
        if self.length == 0:
            print("Stack is Empty")
            return
        print (self.array[self.length - 1])


stack = Stack(5)

stack.push(2)
stack.push(3)
stack.push(4)
stack.push(5)
stack.pop()
stack.peek()

File: stack.js
class Stack {
  constructor(size) {
    this.array = new Array(size);
    this.length = 0;
  }

  push = (data) => {
    if (!data) {
      console.log("Invalid Argument");
      return;
    }
    if (this.length == this.array.length) {
      console.log("Stack Overflow");
      return;
    }
    this.array[this.length] = data;
    console.log(this.array);
    this.length++;
  };

  pop = () => {
    if (this.length == 0) {
      console.log("Cannot pop from empty stack");
      return;
    }
    --this.length;
    let x = this.array[this.length];
    delete this.array[this.length];
    console.log(x);
    console.log(this.array);
    return x;
  };

  printStack = () => {
    console.log(this.array);
  };

  peek = () => {
    if (this.length == 0) {
      console.log("Stack is Empty");
      return;
    }
    console.log(this.array[this.length - 1]);
  };
}

stack = new Stack(5);

stack.push(1);
stack.push(2);
stack.push(3);
stack.push(4);
stack.push(5);
stack.pop();
stack.peek();
